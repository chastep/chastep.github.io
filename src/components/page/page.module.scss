@use "src/assets/styles/mixins";
@use "src/assets/styles/variables";

.page {
  grid-column: span 12;
  @include mixins.margin-bottom(2);

  .inner {
    @include mixins.padding(1, 0.75, 0);
  }

  .title {
    font-weight: 600;
    font-size: variables.$typography-base-font-size * 2.5;
    @include mixins.line-height(2);
    @include mixins.margin-top(0);
    @include mixins.margin-bottom(1.45);
  }

  .body {
    font-size: variables.$typography-base-font-size;
    @include mixins.line-height(1);
    @include mixins.margin(0, 0, 1);
  }
}

@include mixins.breakpoint-sm {
  .page {
    grid-column: span 7;

    .inner {
      @include mixins.padding(1.25, 0.75, 0);
    }
  }
}

@include mixins.breakpoint-lg {
  .page {
    grid-column: span 8;

    .inner {
      @include mixins.padding(1.5, 1, 0);
    }
  }
}
