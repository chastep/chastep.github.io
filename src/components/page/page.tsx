import React, { type FC, type ReactNode } from "react";

import * as styles from "./page.module.scss";

interface PageProps {
  title?: string;
  children: ReactNode;
}

const Page: FC<PageProps> = ({ title, children }) => {
  return (
    <div className={styles.page}>
      <div className={styles.inner}>
        {title && <h1 className={styles.title}>{title}</h1>}
        <div className={styles.body}>{children}</div>
      </div>
    </div>
  );
};

export { Page };
