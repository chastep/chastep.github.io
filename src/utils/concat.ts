const concat = (...args: string[]): string => args.join("");

export { concat };
