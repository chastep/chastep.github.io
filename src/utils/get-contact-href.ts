const getContactHref = (name: string, contact: string) => {
  const hrefs: { [key: string]: string } = {
    mastodon: `${contact}`,
    email: btoa(contact) || "",
    line: `line://ti/p/${contact}`,
    x: `https://www.x.com/${contact}`,
    telegram: `https://t.me/${contact}`,
    vkontakte: `https://vk.com/${contact}`,
    medium: `https://medium.com/${contact}`,
    github: `https://github.com/${contact}`,
    weibo: `https://www.weibo.com/${contact}`,
    gitlab: `https://www.gitlab.com/${contact}`,
    codepen: `https://www.codepen.io/${contact}`,
    soundcloud: `https://soundcloud.com/${contact}`,
    facebook: `https://www.facebook.com/${contact}`,
    instagram: `https://www.instagram.com/${contact}`,
    linkedin: `https://www.linkedin.com/in/${contact}`,
    youtube: `https://www.youtube.com/channel/${contact}`,
    bluesky: `https://bsky.app/profile/${contact}.bsky.social`,
  };

  return hrefs[name] ?? contact;
};

export { getContactHref };
