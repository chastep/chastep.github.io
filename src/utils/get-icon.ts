import { icons } from "@/constants/icons";

const getIcon = (name: keyof typeof icons) => icons[name] || {};

export { getIcon };
