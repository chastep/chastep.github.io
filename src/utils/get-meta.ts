const getMeta = (container: HTMLElement, key: string) => {
  const meta = container.getElementsByTagName("meta");

  for (let i = 0; i < meta.length; i += 1) {
    if (
      key === meta[i].getAttribute("name") ||
      key === meta[i].getAttribute("property")
    ) {
      return meta[i].getAttribute("content");
    }
  }

  return "";
};

export { getMeta };
