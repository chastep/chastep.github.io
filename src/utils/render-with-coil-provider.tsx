import React, { type ReactElement, type ReactNode, type FC } from "react";
import { CoilProvider } from "@alxshelepenok/diesel";
import { render, type RenderOptions } from "@testing-library/react";
import renderer, { type ReactTestRenderer, type TestRendererOptions } from "react-test-renderer";

interface WithCoilProviderProps {
  children: ReactNode;
}

const WithCoilProvider: FC<WithCoilProviderProps> = ({ children }) => (
  <CoilProvider>{children}</CoilProvider>
);

const renderWithCoilProvider = (
  ui: ReactElement,
  options?: RenderOptions,
) => render(ui, { wrapper: WithCoilProvider, ...options });

export const createSnapshotsRenderer = (
  nextElement: ReactElement,
  options?: TestRendererOptions,
): ReactTestRenderer =>
  renderer.create(<CoilProvider>{nextElement}</CoilProvider>, options);

export { renderWithCoilProvider };
