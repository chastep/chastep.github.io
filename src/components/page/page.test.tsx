import React from "react";
import { test, describe, expect } from "bun:test";

import * as mocks from "@/mocks";
import { Page } from "@/components/page/page";
import { createSnapshotsRenderer } from "@/utils/render-with-coil-provider";

describe("Page", () => {
  test("renders correctly", () => {
    const props = {
      children: mocks.markdownRemark.html,
      title: mocks.markdownRemark.frontmatter.title,
    };

    const tree = createSnapshotsRenderer(<Page {...props}>{props.children}</Page>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
