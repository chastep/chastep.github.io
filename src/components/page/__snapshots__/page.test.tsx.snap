// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Page renders correctly 1`] = `
{
  "$$typeof": Symbol(react.test.json),
  "children": [
    {
      "$$typeof": Symbol(react.test.json),
      "children": [
        {
          "$$typeof": Symbol(react.test.json),
          "children": [
            "Perfecting the Art of Perfection",
          ],
          "props": {
            "className": undefined,
          },
          "type": "h1",
        },
        {
          "$$typeof": Symbol(react.test.json),
          "children": [
            "<p>An Essay on Typography by Eric Gill takes the reader back to the year 1930. The year when a conflict between two worlds came to its term. The machines of the industrial world finally took over the handicrafts.</p>",
          ],
          "props": {
            "className": undefined,
          },
          "type": "div",
        },
      ],
      "props": {
        "className": undefined,
      },
      "type": "div",
    },
  ],
  "props": {
    "className": undefined,
  },
  "type": "div",
}
`;
